src/main/java/services/petshop/steps/CommonMethodsPetSteps.java

```diff
@@ -1,4 +1,40 @@
 package services.petshop.steps;
 
+
+import io.restassured.response.Response;
+
+import org.junit.Assert;
+import net.thucydides.core.annotations.Step;
+import net.thucydides.core.annotations.Steps;
+import services.petshop.pojo.requests.Pet;
+
 public class CommonMethodsPetSteps {
+
+    @Steps
+    PostPetSteps postPetSteps;
+
+    @Steps
+    GetPetSteps getPetSteps;
+
+    @Step("When I retrieve the id of the created pet")
+    public void getTheIdOfTheCreatedPet(Pet mypet){
```

Comment:
If you want to create also a pet and not only get an ID for an existing pet, then you should rename the method. But the way you used it, seems you just wanted to get the id of an existing pet, but instead you created a new one. See petshop.post.FlowPetTest#createGetUpdatePet